### DISCUSSION TITLE
`transformers` parameter in generate_pseudo_pop()

### DISCUSSION THREAD

**Discussion by m-qin** [NSAPH-Software/CausalGPS#142] [Q&A] [answered]
Hello! I had a couple clarifications/questions about the `transformers` parameter in generate_pseudo_pop().

1. Can I only enter "pow2" and "pow3" as transformers, or can I also enter common functions like "sqrt" or a function I define beforehand like "logit"? I'm a little confused from the documentation, which says there are [only?] 2 available transformers but users may enter a customized function.
2. If the algorithm uses these transformations to improve covariate balance, are the covariates transformed in the output of generate_pseudo_pop(), which might be called `matched_pop$pseudo_pop`? Are `matched_pop$original_corr_results` and `matched_pop$adjusted_corr_results` evaluated on the transformed covariates? Is it recorded anywhere which transformations were used?

Thanks!

**Comment by Naeemkh** [comment, marked as answer]
You can use any unary function.  You can define them before using them. 

```r
pow5 <- function(x) {x^5}
```
See the `.log` file for details of transformers.

**Comment by ananodehi** [comment]
Hello,
I’ve been having some difficulty applying transformations to the confounding variables in the GPS matching. Specifically, when I use the options use_cov_transform = TRUE and transformers = list("pow2", "pow3", "abs", "scale"), the results appear identical to when use_cov_transform = FALSE.
Could you please help me understand why this might be happening?
Thank you in advance for your assistance.
